dev_guide/builds.adoc

```diff
@@ -846,43 +850,44 @@ source:
 
 == Binary Source
 
-When the `*BuildConfig.spec.source.type*` is `*Binary*`, the build will expect a
-binary as input, and an inline Dockerfile is optional.
-
-The binary is generally assumed to be a tar, gzipped tar, or zip file depending
-on the strategy. For `*Docker*` builds, this is the build context and an
-optional Dockerfile may be specified to override any Dockerfile in the build
-context. For `*Source*` builds, this is assumed to be an archive as described
-above. For `*Source*` and `*Docker*` builds, if `*binary.asFile*` is set the
-build context will consist of a single file named by the value of `*binary.asFile*`. 
-The `*contextDir*` field may be used when an archive is provided. Custom builds will 
-receive this binary as input on standard input (`stdin`).
+Streaming content in binary format from a local file system to the builder is called a `*binary type build*`.
+The corresponding value of `*BuildConfig.spec.source.type*` is `*Binary*` for such builds.
 
-A binary source potentially extracts content (if `*asFile*` is not set), in which case `*contextDir*`
-allows changing to a subdirectory within the content before the build executes.
-
-The source definition is part of the `*spec*` section in the `*BuildConfig*`:
+This source type is unique in that it is leveraged soley based on your use of the `oc start-build`.
 
+[NOTE]
 ====
-[source,yaml]
-----
-source:
-  type: "Binary"
-  binary: <1>
-    asFile: "webapp.war" <2>
-  contextDir: "app/dir" <3>
-  dockerfile: "FROM centos:7\nRUN yum install -y httpd" <4>
-----
-<1> The `*binary*` field specifies the details of the binary source.
-<2> The `*asFile*` field specifies the name of the file that will be created with
-the binary contents.
-<3> The `*contextDir*` field specifies a subdirectory within the extracted contents of a
-binary archive.
-<4> If the optional `*dockerfile*` field is provided, it should be a string
-containing an inline Dockerfile that potentially replaces one within the
-contents of the binary archive.
+Since binary type builds require content to be streamed from the local file system, it is not possible to automatically trigger a binary type build (eg via an image change trigger)
+because the binary files could not be provided. Similarly you cannot launch binary type builds from the web console.
 ====
 
+To utilize binary builds, invoke `oc start-build` with one of these options:
+
+1) `--from-file`:  The contents of the file you specify will be sent as a binary stream to the builder.  Then
+the builder will store the data in a file with the same name at the top of the build context (this internally maps
+to the `*binary.asFile*` field of the `*Binary*` source type).
+
+2) `--from-dir` and `--from-repo`:  The contents will be archived (either zip, tar, or gzipped tar based on what
+is available on the client system) and sent as a binary stream to the builder.  Then the builder will extract
+the contents of the archive within the build context directory.
```

Comment:
Yeah, the code does not leverage a local binary.  I had seen that during a code walkthrough after writing that and had meant to correct this statement, but it slipped through the cracks.

I think the more illustrative lines in startbuild.go are here:  https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/startbuild.go#L418-L486 ... it either tar, tar.gz, or just streams the file bytes based on exactly which option.